module/VuFind/src/VuFind/OAuth2/Entity/UserEntity.php

```diff
@@ -170,31 +180,31 @@ public function getClaims()
                 case 'full_name':
                     // full_name is a special field for firstname + lastname:
                     $result[$claim] = trim(
-                        $this->user['firstname'] . ' ' . $this->user['lastname']
+                        $this->user->getFirstname() . ' ' . $this->user->getLastname()
                     );
                     break;
-                case 'last_language':
+                case 'getLastLanguage':
                     // Make sure any country code is in uppercase:
-                    $value = $this->user->last_language;
+                    $value = $this->user->getLastLanguage();
                     $parts = explode('-', $value);
                     if (isset($parts[1])) {
                         $value = $parts[0] . '-' . strtoupper($parts[1]);
                     }
                     $result[$claim] = $value;
                     break;
                 case 'library_user_id_hash':
-                    $id = $profile['cat_id'] ?? $this->user->cat_username ?? null;
+                    $id = $profile['cat_id'] ?? $this->user->getCatUsername() ?? null;
                     if ($id) {
                         $result[$claim] = hash(
                             'sha256',
-                            $id
-                            . $this->oauth2Config['Server']['hashSalt']
+                            $id . $this->oauth2Config['Server']['hashSalt']
                         );
                     }
                     break;
                 default:
                     if (
-                        ($value = $this->user->{$field} ?? null)
+                        (method_exists($this->user, $field)
```

Comment:
Could you simplify this with the nullsafe operator, e.g. `$this->user?->{$field}() ?? null`?

Reply:
I was not aware that the nullsafe operator also worked with non-existing methods, good to know!

Reply:
The tests seem to fail with `Error: Call to undefined method` trying to use this

Reply:
Yes, I was mistaken about the application of the operator here. Sorry about that!